components/adv-post-form.js

```diff
@@ -30,18 +30,33 @@ export default function AdvPostForm ({ children, item }) {
   const [show, setShow] = useState(undefined)
 
   useEffect(() => {
-    const isDirty = formik?.values.forward?.[0].nym !== '' || formik?.values.forward?.[0].pct !== '' || formik?.values.boost !== '' || (router.query?.type === 'link' && formik?.values.text !== '')
+    const isDirty = formik?.values.forward?.[0].nym !== '' || formik?.values.forward?.[0].pct !== '' ||
+      formik?.values.boost !== '' || (router.query?.type === 'link' && formik?.values.text !== '')
 
+    // if the adv post form is dirty on first render, show the accordian
     if (isDirty) {
-      setShow(false)
+      setShow(true)
     }
-  }, [formik?.values])
 
-  useEffect(() => {
-    const hasBoostError = !!formik?.errors?.boost && formik?.errors?.boost !== ''
-    const hasForwardError = (!!formik?.errors?.forward?.[0].nym && formik?.errors?.forward?.[0].nym !== '') || (!!formik?.errors?.forward?.[0].pct && formik?.errors?.forward?.[0].pct !== '')
+    // HACK ... TODO: we should generically handle this kind of local storage stuff
+    // in the form component, overriding the initial values
+    if (storageKeyPrefix) {
+      for (let i = 0; i < MAX_FORWARDS; i++) {
+        ['nym', 'pct'].forEach(key => {
+          const value = window.localStorage.getItem(`${storageKeyPrefix}-forward[${i}].${key}`)
+          if (value) {
+            formik?.setFieldValue(`forward[${i}].${key}`, value)
+          }
+        })
+      }
+    }
+  }, [formik?.values, storageKeyPrefix])
 
-    setShow(hasForwardError || hasBoostError ? formik?.isSubmitting : undefined)
+  useEffect(() => {
+    // force show the accordian if there is an error and the form is submitting
+    const hasError = !!formik?.errors?.boost || formik?.errors?.forward?.length > 0
+    // if it's open we don't want to collapse on submit
+    setShow(show => show || (hasError && formik?.isSubmitting))
```

Comment:
On testing this branch I am not able to get it to open options on error.  I think we need to cirlce back to this setState.  I agree that the ternary behavior was confusing but it was the only way I get the errors to work.

Reply:
That's true after it's manually closed. I'll fix it.

Reply:
71213179 does the ternary thing but is more explicit. 

The only reason this is needed is because of the `useEffect` dependency which we need. There has to be a better way to do this but it'll have to be part of a future form cleanup.